### PR TITLE
feat(orders): ORDERS-3936 Add consignments specification to order v2 …

### PR DESCRIPTION
**Pull request by dornah**
…api spec.

**Note:** This PR is not ready for Merge as the consignments is still in Beta, rather it's a ground work for when we want to create a DevDoc ticket for the official documentation in the future.

### BOPIS order specs
https://jira.bigcommerce.com/browse/PROJECT-3381

### POST
Screen Shot 2021-04-30 at 7 45 18 am

Screen Shot 2021-04-30 at 7 47 05 am

Screen Shot 2021-04-30 at 7 46 19 am

### GET all consignments
Screen Shot 2021-04-30 at 7 50 33 am

### GET single consignment
Screen Shot 2021-04-30 at 7 51 35 am

### PUT
Screen Shot 2021-04-30 at 7 55 26 am

ping @bigcommerce/orders

### CODE REVIEW

File: reference/orders.v2.oas2.yml [hunk 1]
```diff
@@ -4412,13 +4620,229 @@ components:
         - type: object
           properties:
             products:
-              type: array
-              items:
-                anyOf:
-                  - $ref: '#/components/schemas/orderCustomProduct_Post'
-                  - $ref: '#/components/schemas/orderCatalogProduct_Post'
+              $ref: '#/components/schemas/orderProduct_Post'
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:4623]: we may need to update the line "description: Products and Billing address only required for POST operation." to mention about either "consignments" or "products" field is required

File: reference/orders.v2.oas2.yml [hunk 2]
```diff
@@ -347,10 +366,63 @@ paths:
                           value: '180'
                         - id: 230
                           value: '192'
+              Pickup Order:
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:369]: we may need to update "description" of "post" about the "required fields": either "consignments" or "products" field is required

File: reference/orders.v2.oas2.yml [hunk 3]
```diff
@@ -347,10 +366,63 @@ paths:
...
+                    city: Austin
+                    state: Texas
+                    zip: '78751'
+                    country: United States
+                    country_iso2: US
+                    email: janedoe@example.com
         required: true
       responses:
-        '200':
+        '201':
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:396]: 👏

File: reference/orders.v2.oas2.yml [hunk 4]
```diff
@@ -4412,13 +4620,229 @@ components:
...
+                      type: object
+                      description: |-
+                        - The default values for optional fields come from the pickup store location specified via id field. Use the optional fields for the values you need to overwrite.
+
+                        - If inventory is being tracked on an item, there should be sufficient stock for the item in the pickup location, and the related inventory will be deducted upon successful creation.
+                      properties:
+                        id:
+                          type: integer
+                      required:
+                        - id
```
Comment (donald-nguyen-bc) [inline reference/orders.v2.oas2.yml:4666]: This seems to be duplicates of orderPickup, should we merge the "description" and "required" info to orderPickup model ?

**Review by donald-nguyen-bc — APPROVED** [review]
👏

File: reference/orders.v2.oas2.yml (hunk 4 above)
Comment (dornah) [reply reference/orders.v2.oas2.yml:4666]: My intention was to distinguish between required fields and description of POST and PUT:

location id is not required for PUT and the  description for PUT is:

- If pickup location id is updated, rest of the location fields will be updated to reflect the new location information. If you need to overwrite these values you should provide them in the payload.
                            
- If inventory is being tracked on an item, there should be sufficient stock for the item in the new pickup location, and the related inventory will be deducted from the new location and return to the old location upon successful update.

File: reference/orders.v2.oas2.yml (hunk 2 above)
Comment (dornah) [reply reference/orders.v2.oas2.yml:369]: Good catch. Does this address your comment?
https://github.com/bigcommerce/api-specs/pull/315/commits/63628293af4fe7d659707f100c2e34352f309b5d

File: reference/orders.v2.oas2.yml (hunk 1 above)
Comment (dornah) [reply reference/orders.v2.oas2.yml:4623]: https://github.com/bigcommerce/api-specs/pull/315/files#r623521605

File: reference/orders.v2.oas2.yml (hunk 4 above)
Comment (donald-nguyen-bc) [reply reference/orders.v2.oas2.yml:4666]: ah I see, TIL about that trick, it' really smart 😄. Thanks for explanation Dorna 🙇

Comment (dornah) [reply reference/orders.v2.oas2.yml:4666]: Haha, nw 🙇‍♀️
It's a bit messy in the Yaml format, but the rendered result is good.

File: reference/orders.v2.oas2.yml (hunk 2 above)
Comment (donald-nguyen-bc) [reply reference/orders.v2.oas2.yml:369]: 👍

File: reference/orders.v2.oas2.yml (hunk 1 above)
Comment (donald-nguyen-bc) [reply reference/orders.v2.oas2.yml:4623]: 👍

**Comment by aglensmith** [comment]
Ready to be merged?

**Pull request merged**